src/commands/handlers/registration.js

```diff
@@ -144,7 +144,12 @@ var handlers = {
                 }
                 if (this.network.cap.enabled.length > 0) {
                     if (this.network.cap.isEnabled('sasl')) {
-                        this.connection.write('AUTHENTICATE PLAIN');
+                        if (this.connection.options.sasl_mechanism && 
+                            this.connection.options.sasl_mechanism == "AUTHCOOKIE") {
```

Comment:
This if statement should be rewritten as: `if (this.connection.options.sasl_mechanism === 'AUTHCOOKIE') {`

Reply:
Good point! I've update the commit.